modules/api/src/main/java/org/apache/fluo/api/config/SimpleConfiguration.java

```diff
@@ -250,6 +277,22 @@ public SimpleConfiguration subset(String prefix) {
     return new SimpleConfiguration(internalConfig.subset(prefix));
   }
 
+  /**
+   * @param SimpleConfigs SimpleConfiguration's to be merged
+   * @return SimpleConfiguration
```

Comment:
I think this describes the behaviour of `addConfiguration()` in CompositeConfiguration. You are currently using `addProperty` which will create a list of values for properties in multiple files.  Best way to confirm any behaviour is to just create a simple unit test.